lib/srv/alpnproxy/listener.go

```diff
@@ -88,7 +90,9 @@ func (l *ListenerMuxWrapper) startAcceptingConnectionServiceListener() {
 	for {
 		conn, err := l.Listener.Accept()
 		if err != nil {
-			l.errC <- err
+			if !utils.IsUseOfClosedNetworkError(err) {
+				l.errC <- err
```

Comment:
The `IsUseOfClosedNetworkError` error will not be propagated to the `ListenerMuxWrapper` consumer - wonder why we want that. For example if `ListenerMuxWrapper.Close` function was not called the consumer loop will stuck at  `*ListenerMuxWrapper) Accept()` function and block goroutine.

Reply:
If `ListenerMuxWrapper.Close` was not called then we can still call `HandleConnection` on it and add a new connection that should be handled by the `Accept` loop, right? If we want to piggyback on the inner listener to eventually break us out of accept loops we should make it explicit - especially because we do create the occasional `ListenerMuxWrapper` with a `nil` listener, intended only for use with `HandleConnection`.

Perhaps the solution is to use a context that closes the `ListenerMuxWrapper` when done, like `lib/multiplexer.Listener`?

Reply:
Yea, canceling the context should work: 

Right now the  `ListenerMuxWrapper.Close()` function is called on `OnExit` flow  but in case of `TeleportReloadEvent` it might now work so we should cover this case.

Reply:
After a `TeleportReloadEvent` - and after the new `TeleportProcess` reports it's ready - we do a regular `Shutdown()`, so as long as every service closes its "inner" listeners in the `OnExit` (which, as far as I can tell, every service does) and we close the outer listeners on shutdown (done in this PR) I think we're cleaning up things correctly - except maybe during a fast close, but the process is going to exit ASAP in that case, so who cares.